File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
const solution = (sizes) => {
    const [hor, ver] = sizes.reduce(([mw, mh], [w, h]) => [Math.max(mw, Math.max(w, h)), Math.max(mh, Math.min(w, h))], [0, 0] )
    return hor * ver
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_big, max_small = 0, 0
    for size in sizes:
        bigger = max(size[0], size[1])
        smaller = min(size[0], size[1])
        max_big = max(bigger, max_big)
        max_small = max(smaller, max_small)
    return max_big * max_small